### PR TITLE
Support multiple branches

### LINKED ISSUE #158: Add support for testing multiple branches
Now the server jar/docker image is not a command line argument, making it hard to test against different branches of the server.  Create a new command line argument so this can be overridden.

### PR DESCRIPTION
**Pull request by tzolov**
Resolves #158

 - Add new command line arguments:
   -sv | --skipperVersion - set the skipper version to test (e.g. 1.0.4.BUILD-SNAPSHOT)
   -dv | --dataflowVersion - set the dataflow version to test (e.g. 1.5.0.BUILD-SNAPSHOT)
   -av | --appsVersion - set the stream app version to test (e.g. Celsius.SR2). Apps should be accessible via maven repo or docker hub.
   -tv | --tasksVersion - set the task app version to test (e.g. Clark.RELEASE). Tasks should be accessible via maven repo or docker hub.
 - map the new arguments to the SKIPPER_VERSION, DATAFLOW_VERSION, STREAM_APPS_VERSION, TASKS_VERSION env. variables
 - alter the env.properties to add versions resolved from the env. variables
 - fix problem in run.sh to set the enableStreamFeatrues to true
 - remove the deprecated ${random-value} form the generated manifest and use the bash's $RANDOM expression instead.
 - generate the skipper.yml for kubernetes (similar to the kubernetes server yml generation).

### CODE REVIEW

**Comment by markpollack** [comment]
squashed and merged as f040852ff7844a64e9973933d55263fee0aaccde

File: kubernetes/server/create.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
     spec:
       containers:
       - name: scdf
-        image: springcloud/spring-cloud-dataflow-server-kubernetes:latest
+        image: springcloud/spring-cloud-dataflow-server-kubernetes:$DATAFLOW_VERSION
```
Comment (chrisjs) [inline kubernetes/server/create.sh]: For docker images we publish milestone, release candidate and release artifacts in the typical format (https://hub.docker.com/r/springcloud/spring-cloud-dataflow-server-kubernetes/tags/) but we don't publish explicit SNAPSHOT versions. When the default are used, an image by name of`1.5.0.BUILD-SNAPSHOT` is attempted to be fetched which doesn't exist. When master is built in CI we publish to a docker tag `latest` which represents the current master "SNAPSHOT".

I think in this case we want to check the `$DATAFLOW_VERSION` and if its a `SNAPSHOT` change the tag to `latest`. We do a similar manipulation in the documentation where we align image names to versions: https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-docs/pom.xml#L325

File: kubernetes/skipper-server/create.sh [hunk 2]
```diff
@@ -14,7 +14,72 @@ function kubectl_create() {
...
+  selector:
+    name: skipper
+  template:
+    metadata:
+      labels:
+        name: skipper
+    spec:
+      containers:
+      - name: skipper
+        image: springcloud/spring-cloud-skipper-server:$SKIPPER_VERSION
```
Comment (chrisjs) [inline kubernetes/skipper-server/create.sh]: i think the same would apply with the comment in `kubernetes/server/create.sh`

**Comment by chrisjs** [comment]
1 - I was able to run combinations of SCDF master / SCDF 1.4 against rabbit and kafka in `classic` mode

2 - I was not able to run any combination of SCDF / Skipper against kafka in `skipper` mode as I would find in launched application logs, for example:

`Caused by: org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 10000`

3 - Similar error when using RabbitMQ in `skipper` yet starting with boot not being able to convert the Rabbit port

I noticed `kubernetes/skipper-server/skipper.yml` gets modified on disk and seems some vars may not be getting replaced properly (or my setup is invalid):

```value: "{\"spring.cloud.skipper.server.enableLocalPlatform\" : false, \"spring.cloud.skipper.server.platform.kubernetes.accounts.cluster1.environmentVariables\" : \"SPRING_RABBITMQ_HOST=,SPRING_RABBITMQ_PORT=,SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS=:,SPRING_CLOUD_STREAM_KAFKA_BINDER_ZK_NODES=:2181\",\"spring.cloud.skipper.server.platform.kubernetes.accounts.cluster1.memory\" : \"1024Mi\",\"spring.cloud.skipper.server.platform.kubernetes.accounts.cluster1.createDeployment\" : true}"```

File: kubernetes/skipper-server/create.sh (hunk 2 above)
Comment (tzolov) [reply kubernetes/skipper-server/create.sh]: Indeed there are no `latest` in the skipper (perhaps @markpollack knows why?) but it fits well current acceptance tests model :)

**Comment by chrisjs** [comment]
closing - added escaping in 4027bb7

**Pull request closed without merging**